File: pet-service/src/main/java/com/phorus/petservice/model/dtos/PetDTO.java
package com.phorus.petservice.model.dtos;

import com.phorus.petservice.model.validationGroups.PetValidationGroup;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.experimental.SuperBuilder;

import javax.validation.constraints.Min;
import javax.validation.constraints.NotBlank;
import javax.validation.constraints.NotNull;
import javax.validation.constraints.Size;


@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@SuperBuilder
public class PetDTO {

    @NotBlank(groups = PetValidationGroup.PetCreateValGroup.class, message = "Cannot be blank")
    @Size(groups = {PetValidationGroup.PetCreateValGroup.class,
            PetValidationGroup.PetUpdateValGroup.class},
            min = 3,
            message = "Must have at least 3 characters long")
    private String name;

    @NotNull(groups = PetValidationGroup.PetCreateValGroup.class, message = "Cannot be null")
    @Min(groups = {PetValidationGroup.PetCreateValGroup.class,
            PetValidationGroup.PetUpdateValGroup.class},
            value = 0,
            message = "Cannot be negative")
    private Integer age;

    @NotNull(groups = PetValidationGroup.PetCreateValGroup.class, message = "Cannot be null")
    private Long userId; // Owner
}

File: pet-service/src/main/kotlin/com/phorus/petservice/model/dtos/PetDTO.kt
package com.phorus.petservice.model.dtos

import com.phorus.petservice.model.validationGroups.PetCreateValGroup
import com.phorus.petservice.model.validationGroups.PetUpdateValGroup

import javax.validation.constraints.Min
import javax.validation.constraints.NotBlank
import javax.validation.constraints.NotNull
import javax.validation.constraints.Size

data class PetDTO(
    @field:NotBlank(groups = [PetCreateValGroup::class], message = "Cannot be blank")
    @field:Size(
        groups = [PetCreateValGroup::class, PetUpdateValGroup::class],
        min = 3,
        message = "Must have at least 3 characters long"
    )
    var name: String? = null,

    @field:NotNull(groups = [PetCreateValGroup::class], message = "Cannot be null")
    @field:Min(
        groups = [PetCreateValGroup::class, PetUpdateValGroup::class],
        value = 0,
        message = "Cannot be negative"
    )
    var age: Int? = null,

    @field:NotNull(groups = [PetCreateValGroup::class], message = "Cannot be null")
    var userId: Long? = null,
)

data class PetResponse(
    var id: Long? = null,
    var name: String? = null,
    var age: Int? = null,
    var userId: Long? = null,
)